src/Controls/tests/Core.UnitTests/AppThemeTests.cs

```diff
@@ -230,19 +324,92 @@ public void NullApplicationCurrentFallsBackToEssentials()
 
 			Application.Current = null;
 
+			app.LoadPage(new ContentPage { Content = label });
+
 			Assert.Equal(Colors.Green, label.TextColor);
 
-			SetAppTheme(AppTheme.Dark);
+			EmulatePlatformThemeChange(AppTheme.Dark);
 
 			Assert.Equal(Colors.Red, label.TextColor);
 		}
 
+		[Fact]
+		public void NullApplicationCurrentFallsBackToEssentialsNonVisualElement()
+		{
+			var element = new NonVisualElement
+			{
+				Text = "Green on Light, Red on Dark"
+			};
+
+			element.SetAppThemeColor(NonVisualElement.ColorProperty, Colors.Green, Colors.Red);
+
+			Application.Current = null;
+
+			Assert.Equal(Colors.Green, element.Color);
+
+			EmulatePlatformThemeChange(AppTheme.Dark);
+
+			Assert.Equal(Colors.Red, element.Color);
+		}
+
 		[Fact]
 		//https://github.com/dotnet/maui/issues/17478
 		public void BindingConversion()
 		{
 			var border = new Border();
 			border.SetAppTheme(Border.StrokeProperty, Colors.Red, Colors.Black);
 		}
+
+		[Fact, Category(TestCategory.Memory)]
+		public async Task VisualElementDoesNotLeak()
+		{
+			(WeakReference Label, WeakReference Binding) CreateReference()
+			{
+				var element = new Label { Text = "Green on Light, Red on Dark" };
+
+				var binding = new AppThemeBinding { Light = Colors.Green, Dark = Colors.Red };
+
+				element.SetBinding(Label.TextColorProperty, binding);
+
+				Assert.True(binding.IsApplied);
+
+				return (new WeakReference(element), new WeakReference(binding));
+			}
+
+			var (element, binding) = CreateReference();
+
+			// GC
+			await TestHelpers.Collect();
+
+			Assert.False(element.IsAlive, "Label should not be alive!");
+			Assert.False(binding.IsAlive, "AppThemeBinding should not be alive!");
+		}
```

Comment:
This test specifically test to see if the GC can pick up a `VisualElement` and `AppThemeBinding` that are connected. Even though the binding only has a weak reference to the target element, the event that it attaches is _not_ weak and _could_leak. But, I do not think it actually does. @jonathanpeppers is this test valid for testing this type of leak?